sdk/containers/azcontainerregistry/blob_custom_client.go

```diff
@@ -79,6 +81,19 @@ func NewBlobDigestCalculator() *BlobDigestCalculator {
 	}
 }
 
+func (b *BlobDigestCalculator) saveState() error {
```

Comment:
Since all our SDK APIs does not contains state, if we want to provide sha256 sum function for SDK user for the blob uploading, we need user to maintain and pass the correct calculator. See example here: https://github.com/Azure/azure-sdk-for-go/blob/77eb24da2f78811cab080bf5c5cb7d4e3a7222a9/sdk/containers/azcontainerregistry/example_upload_download_blob_test.go#L34-L42.

Reply:
One more thing, do you think we should export `saveState` and `restoreState`?